src/wp-includes/global-styles-and-settings.php

```diff
@@ -257,21 +257,17 @@ function wp_add_global_styles_for_blocks() {
 	$block_nodes = $tree->get_styles_block_nodes();
 
 	$can_use_cached = ! wp_is_development_mode( 'theme' );
+
 	if ( $can_use_cached ) {
-		// Hash global settings and block nodes together to optimize performance of key generation.
-		$hash = md5(
-			wp_json_encode(
-				array(
-					'global_setting' => wp_get_global_settings(),
-					'block_nodes'    => $block_nodes,
-				)
-			)
-		);
-
-		$cache_key = "wp_styles_for_blocks:$hash";
-		$cached    = get_site_transient( $cache_key );
-		if ( ! is_array( $cached ) ) {
-			$cached = array();
+		// Hash the merged WP_Theme_JSON data to bust cache on settings or styles change.
+		$cache_hash = md5( wp_json_encode( $tree->get_raw_data() ) );
```

Comment:
This is a much simpler hash generation, and accounts for a much broader set of changes that might happen to the merged Theme JSON data, which could affect global block styles, which is less risky.